### PR TITLE
Persist `Environment` in metadata

### LINKED ISSUE #1695: Persist `Environment` types and values in metadata
In [this section](https://github.com/paritytech/ink/blob/master/ARCHITECTURE.md#environment-trait) of our `ARCHITECTURE.md` we explain what the `Environment` trait in ink! is used for. The `DefaultEnvironment` which ink! uses is defined [here](https://github.com/paritytech/ink/blob/master/crates/env/src/types.rs#L169-L183).

Right now, we don't persist which `Environment` implementation the contract uses anywhere. This means someone can deploy a contract with types that are different from those a chain uses to that chain. In https://github.com/paritytech/ink/issues/1694 we will investigate the criticality of this issue. But whatever turns out there, we should have a feature that warns contract developers about this. It is a serious oversight.

## ToDo
- [x] Persist the `Environment` types and `MAX_EVENT_TOPICS` value to the metadata. Will have to be a new field `environment`.
- [x] Make sure the schema and formal definition under `metadata/` is up to date.
- [x] Create a follow-up PR to `ink-docs` to also update the documentation on metadata there.
- [x] Talk to @statictype about where to create a follow-up issue for the UI to validate that the chain's types match the environment denoted in the metadata. Either `polkadot-js/api` or both `polkadot-js/apps` and Contracts UI. I'm not sure where this validation should happen.
- [x] Create a follow-up issue for `cargo-contract` to validate that the chain's types match the environment denoted in the metadata.

### PR DESCRIPTION
**Pull request by SkymanOne**
This PR closes #1695 by adding a field `environment` to `metadata.json` which contains the info about the types of the contract's environment.

### CODE REVIEW

File: crates/env/Cargo.toml [hunk 1]
```diff
@@ -16,7 +16,6 @@ categories = ["no-std", "embedded"]
 include = ["Cargo.toml", "src/**/*.rs", "README.md", "LICENSE"]
 
 [dependencies]
-ink_metadata = { version = "4.1.0", path = "../metadata", default-features = false, features = ["derive"], optional = true }
```
Comment (SkymanOne) [inline crates/env/Cargo.toml:19]: Removed this because it seems it is not used anywhere.

Comment (SkymanOne) [inline crates/env/Cargo.toml:19]: Removed this because it seems it is not used anywhere.

File: crates/metadata/src/specs.rs [hunk 2]
```diff
@@ -59,6 +62,7 @@ pub struct ContractSpec<F: Form = MetaForm> {
     docs: Vec<F::String>,
     /// The language specific error type.
     lang_error: TypeSpec<F>,
+    environment: EnvironmentSpec<F>,
```
Comment (cmichi) [inline crates/metadata/src/specs.rs:65]: Comment plz.

File: crates/metadata/src/specs.rs [hunk 3]
```diff
@@ -115,6 +121,10 @@ where
     pub fn lang_error(&self) -> &TypeSpec<F> {
         &self.lang_error
     }
+
+    pub fn environment(&self) -> &EnvironmentSpec<F> {
```
Comment (cmichi) [inline crates/metadata/src/specs.rs:125]: Comment plz.

**Review by cmichi — APPROVED** [review]

**Pull request merged**